src/bin/casr-python.rs

```diff
@@ -0,0 +1,296 @@
+                .help("Stdin file for program"),
+        )
+        .arg(
+            Arg::new("ARGS")
+                .multiple_values(true)
+                .takes_value(true)
+                .last(true)
+                .help("Add \"-- <path> <arguments>\" to run"),
+        )
+        .get_matches();
+
+    // Get program args.
+    let argv: Vec<&str> = if let Some(args) = matches.values_of("ARGS") {
+        args.collect()
+    } else {
+        bail!("Wrong arguments for starting program");
+    };
+
+    // Get stdin for target program.
+    let stdin_file = if let Some(path) = matches.value_of("stdin") {
+        let file = PathBuf::from(path);
+        if file.exists() {
+            Some(file)
+        } else {
+            bail!("Stdin file not found: {}", file.display());
+        }
+    } else {
+        None
+    };
+
+    // Run program.
+    let mut python_cmd = Command::new(argv[0]);
+    if let Some(ref file) = stdin_file {
+        python_cmd.stdin(std::fs::File::open(file).unwrap());
+    }
+    if argv.len() > 1 {
+        python_cmd.args(&argv[1..]);
+    }
+    let python_result = python_cmd
+        .output()
+        .with_context(|| "Couldn't run target program")?;
+
+    let python_stderr = String::from_utf8_lossy(&python_result.stderr);
+
+    // Create report.
+    let mut report = CrashReport::new();
+    report.executable_path = argv[0].to_string();
+    report.proc_cmdline = argv.join(" ");
+    let _ = report.add_os_info();
+
+    // Get python report.
+    let python_stderr_list: Vec<String> =
+        python_stderr.split('\n').map(|l| l.to_string()).collect();
+
+    if python_stderr_list
+        .iter()
+        .any(|line| line.contains("ERROR: libFuzzer: "))
+    {
+        let python_stdout = String::from_utf8_lossy(&python_result.stdout);
+        let python_stdout_list: Vec<String> =
+            python_stdout.split('\n').map(|l| l.to_string()).collect();
+
+        if let Some(report_start) = python_stdout_list
+            .iter()
+            .position(|line| line.contains("Uncaught Python exception: "))
+        {
+            // Set python report in casr report.
+            let report_end = python_stdout_list
+                .iter()
+                .rposition(|s| !s.is_empty())
+                .unwrap()
+                + 1;
+            report.python_report = Vec::from(&python_stdout_list[report_start..report_end]);
+
+            // Get exception from python report.
+            let re = Regex::new(r#"([\w]+): (.+)"#).unwrap();
+            if let Some(cap) = re.captures(&report.python_report[1]) {
+                report.execution_class.severity = Cow::Borrowed("UNDEFINED");
+                report.execution_class.short_description =
+                    Cow::Borrowed(cap.get(1).unwrap().as_str());
+                report.execution_class.description = Cow::Borrowed(cap.get(2).unwrap().as_str());
+                report.execution_class.explanation = Cow::Borrowed("");
+            }
```

Comment:
This code repeats twice, we should make a function

Reply:
Done